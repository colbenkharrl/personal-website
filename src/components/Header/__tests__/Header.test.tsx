import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../__stories__/Header.stories';

const { Homepage, NonHomepage } = composeStories(stories);

jest.mock('../../Avatar', () => ({
  Avatar: () => <p>Avatar</p>,
}));
jest.mock('../../AvatarContainer', () => ({
  AvatarContainer: (props: object) => <p {...props}>AvatarContainer</p>,
}));
jest.mock('../../MobileNavigation', () => ({
  MobileNavigation: () => <p>MobileNavigation</p>,
}));
jest.mock('../../DesktopNavigation', () => ({
  DesktopNavigation: () => <p>DesktopNavigation</p>,
}));

it('Header matches snapshot (homepage) (regression test)', () => {
  render(<Homepage />);

  expect(screen.getByTestId('Header')).toMatchSnapshot(
    'Full component snapshot.',
  );
});

it('Header renders special avatar on homepage', () => {
  render(<Homepage />);

  screen.getByTestId('Header-home-avatar');
});

it('Header matches snapshot (non-homepage) (regression test)', () => {
  render(<NonHomepage />);
  expect(screen.getByTestId('Header')).toMatchSnapshot(
    'Full component snapshot.',
  );
});

it("Header doesn't renders special avatar on non-homepage", () => {
  render(<NonHomepage />);

  expect(screen.queryByTestId('Header-home-avatar')).toBeNull();
  screen.getByTestId('Header-non-home-avatar');
});
