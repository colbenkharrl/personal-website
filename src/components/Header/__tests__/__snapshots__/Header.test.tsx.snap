// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot (homepage) (regression test): Full component snapshot. 1`] = `
<header
  class="pointer-events-none relative z-50 flex flex-none flex-col"
  data-testid="Header"
>
  <div
    class="order-last mt-[calc(theme(spacing.16)-theme(spacing.3))]"
  />
  <div
    class="sm:px-8 top-0 order-last -mb-3 pt-3"
    data-testid="Container"
    style="position: var(--header-position);"
  >
    <div
      class="mx-auto w-full max-w-7xl lg:px-8"
    >
      <div
        class="relative px-4 sm:px-8 lg:px-12"
      >
        <div
          class="mx-auto max-w-2xl lg:max-w-5xl"
        >
          <div
            class="top-[var(--avatar-top,theme(spacing.3))] w-full"
            data-testid="Header-home-avatar"
            style="position: var(--header-inner-position);"
          >
            <div
              class="relative"
            >
              <p
                class="absolute left-0 top-3 origin-left transition-opacity"
                style="transform: var(--avatar-border-transform);"
              >
                AvatarContainer
              </p>
              <p>
                Avatar
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="top-0 z-10 h-16 pt-6"
    style="position: var(--header-position);"
  >
    <div
      class="sm:px-8 top-[var(--header-top,theme(spacing.6))] w-full"
      data-testid="Container"
      style="position: var(--header-inner-position);"
    >
      <div
        class="mx-auto w-full max-w-7xl lg:px-8"
      >
        <div
          class="relative px-4 sm:px-8 lg:px-12"
        >
          <div
            class="mx-auto max-w-2xl lg:max-w-5xl"
          >
            <div
              class="relative flex gap-4"
            >
              <div
                class="flex flex-1"
              />
              <div
                class="flex flex-1 justify-end md:justify-center"
              >
                <p>
                  MobileNavigation
                </p>
                <p>
                  DesktopNavigation
                </p>
              </div>
              <div
                class="flex justify-end md:flex-1"
              >
                <div
                  class="pointer-events-auto"
                >
                  <button
                    aria-label="Switch to dark theme"
                    class="group rounded-full bg-white/90 px-3 py-2 shadow-lg shadow-zinc-800/5 ring-1 ring-zinc-900/5 backdrop-blur transition dark:bg-zinc-800/90 dark:ring-white/10 dark:hover:ring-white/20"
                    data-testid="ThemeToggle"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-6 w-6 fill-zinc-100 stroke-zinc-500 transition group-hover:fill-zinc-200 group-hover:stroke-zinc-700 dark:hidden [@media(prefers-color-scheme:dark)]:fill-orange-50 [@media(prefers-color-scheme:dark)]:stroke-orange-500 [@media(prefers-color-scheme:dark)]:group-hover:fill-orange-50 [@media(prefers-color-scheme:dark)]:group-hover:stroke-orange-600"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8 12.25A4.25 4.25 0 0 1 12.25 8v0a4.25 4.25 0 0 1 4.25 4.25v0a4.25 4.25 0 0 1-4.25 4.25v0A4.25 4.25 0 0 1 8 12.25v0Z"
                      />
                      <path
                        d="M12.25 3v1.5M21.5 12.25H20M18.791 18.791l-1.06-1.06M18.791 5.709l-1.06 1.06M12.25 20v1.5M4.5 12.25H3M6.77 6.77 5.709 5.709M6.77 17.73l-1.061 1.061"
                        fill="none"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="hidden h-6 w-6 fill-zinc-700 stroke-zinc-500 transition dark:block [@media(prefers-color-scheme:dark)]:group-hover:stroke-zinc-400 [@media_not_(prefers-color-scheme:dark)]:fill-orange-400/10 [@media_not_(prefers-color-scheme:dark)]:stroke-orange-500"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17.25 16.22a6.937 6.937 0 0 1-9.47-9.47 7.451 7.451 0 1 0 9.47 9.47ZM12.75 7C17 7 17 2.75 17 2.75S17 7 21.25 7C17 7 17 11.25 17 11.25S17 7 12.75 7Z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header matches snapshot (non-homepage) (regression test): Full component snapshot. 1`] = `
<header
  class="pointer-events-none relative z-50 flex flex-none flex-col"
  data-testid="Header"
>
  <div
    class="top-0 z-10 h-16 pt-6"
    style="position: var(--header-position);"
  >
    <div
      class="sm:px-8 top-[var(--header-top,theme(spacing.6))] w-full"
      data-testid="Container"
      style="position: var(--header-inner-position);"
    >
      <div
        class="mx-auto w-full max-w-7xl lg:px-8"
      >
        <div
          class="relative px-4 sm:px-8 lg:px-12"
        >
          <div
            class="mx-auto max-w-2xl lg:max-w-5xl"
          >
            <div
              class="relative flex gap-4"
            >
              <div
                class="flex flex-1"
              >
                <p
                  data-testid="Header-non-home-avatar"
                >
                  AvatarContainer
                </p>
              </div>
              <div
                class="flex flex-1 justify-end md:justify-center"
              >
                <p>
                  MobileNavigation
                </p>
                <p>
                  DesktopNavigation
                </p>
              </div>
              <div
                class="flex justify-end md:flex-1"
              >
                <div
                  class="pointer-events-auto"
                >
                  <button
                    aria-label="Switch to dark theme"
                    class="group rounded-full bg-white/90 px-3 py-2 shadow-lg shadow-zinc-800/5 ring-1 ring-zinc-900/5 backdrop-blur transition dark:bg-zinc-800/90 dark:ring-white/10 dark:hover:ring-white/20"
                    data-testid="ThemeToggle"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-6 w-6 fill-zinc-100 stroke-zinc-500 transition group-hover:fill-zinc-200 group-hover:stroke-zinc-700 dark:hidden [@media(prefers-color-scheme:dark)]:fill-orange-50 [@media(prefers-color-scheme:dark)]:stroke-orange-500 [@media(prefers-color-scheme:dark)]:group-hover:fill-orange-50 [@media(prefers-color-scheme:dark)]:group-hover:stroke-orange-600"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8 12.25A4.25 4.25 0 0 1 12.25 8v0a4.25 4.25 0 0 1 4.25 4.25v0a4.25 4.25 0 0 1-4.25 4.25v0A4.25 4.25 0 0 1 8 12.25v0Z"
                      />
                      <path
                        d="M12.25 3v1.5M21.5 12.25H20M18.791 18.791l-1.06-1.06M18.791 5.709l-1.06 1.06M12.25 20v1.5M4.5 12.25H3M6.77 6.77 5.709 5.709M6.77 17.73l-1.061 1.061"
                        fill="none"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="hidden h-6 w-6 fill-zinc-700 stroke-zinc-500 transition dark:block [@media(prefers-color-scheme:dark)]:group-hover:stroke-zinc-400 [@media_not_(prefers-color-scheme:dark)]:fill-orange-400/10 [@media_not_(prefers-color-scheme:dark)]:stroke-orange-500"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17.25 16.22a6.937 6.937 0 0 1-9.47-9.47 7.451 7.451 0 1 0 9.47 9.47ZM12.75 7C17 7 17 2.75 17 2.75S17 7 21.25 7C17 7 17 11.25 17 11.25S17 7 12.75 7Z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
